### ISSUE TITLE
PyDMEnumButton - Margins/Padding too big

### ISSUE THREAD

**Issue by hhslepicka** [slaclab/pydm#683]
**Describe the bug**
When using PyDMEnumButton with AlarmSensitiveBorder, the border is too far from the buttons.
![image](https://user-images.githubusercontent.com/8185425/89945727-a66f0780-dbd6-11ea-8840-3b67789a7fe6.png)

**Expected behavior**
Alarm border closer to the buttons

**Steps to Reproduce**
Use a PyDMEnumButton with alarm sensitive Border and make it cause the channel to go into an alarm state.

**Comment by cristinasewell** [comment]
@hhslepicka I think you were correct - this is not really an issue - It is basically a combination of a couple of things that I did there to cause that behavior.... but we might be able to improve it?? i am not sure....

1. I set the margins of the widget to 0 (displayed in the picture below). 

![all_0s](https://user-images.githubusercontent.com/62306310/91248746-369a6a00-e70a-11ea-87e5-8330a7fd35f8.PNG)

If I set them all to 0s, it looks like in the picture below:

![0s](https://user-images.githubusercontent.com/62306310/91250085-056f6900-e70d-11ea-9025-9e13f79b0c65.PNG)

If I put them back to their default values:

![default_mux](https://user-images.githubusercontent.com/62306310/91250232-5aab7a80-e70d-11ea-80ed-3bc34b5279eb.PNG)

2. Secondly, I am using this in a QGridLayout, the widget is stretching to occupy all the space in the grid, that's why we see the extra padding, but the QPushButtons in the PyDMEnumButton are not necessarily stretching to occupy all the space in the widget itself, by default they don't have their sizePolicy to Expand,  if we set it to expand, it will solve this padding issue - but it could mess up with some existing PyDM screens already, I don't know if we want to do that. 

Here is an example with ` w.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)` and all the margins (designer properties) set to 2: 

![what_we_want](https://user-images.githubusercontent.com/62306310/91255092-d65ef480-e718-11ea-95a6-f9259781ee35.PNG)

which looks great, but I can see how some people's screens might look weird if we change this (depending on their settings), just like this one here: 

![what_happens](https://user-images.githubusercontent.com/62306310/91255586-fc38c900-e719-11ea-8513-a512be02560c.PNG)


-------------------------

There is one more thing, as soon as I was adding an EnumButton to a layout with qt designer - it would expand to its ` minimumSizeHint-> return QSize(50, 100) `  (if i did not specify a set size) instead of trying to stay in the layout just like the other little widgets, and I would make the layout smaller, and it would work fine until I opened the .ui file again with designer: 

Set the grid smaller: 
![designer2](https://user-images.githubusercontent.com/62306310/91253493-d0671480-e714-11ea-95a2-f530aec01bb8.PNG)

Open this .ui file again with qtDesigner - and it resizes again: 
![designer](https://user-images.githubusercontent.com/62306310/91253491-ce9d5100-e714-11ea-87c0-4a2c6e69cf7f.PNG)

But all this can be adjusted if we set the Max and Min size of the widgets from qtDesigner. 
Or, we can return a smaller minimumSizeHint -> maybe `QSize(32, 23)` or something like that, so if we don't set a Maximum size to this widget it does not expand all the way to 100 height.  

-----------------
The bottom line, If we were to do something about this Issue, which I don't know if it would be for the better,  I would suggest we change a couple of  things: 
 `def minimumSizeHint(self): return QSize(32, 23)` or some other smaller than the current numbers
`def generate_widgets(items): -> w.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)`
and maybe set the `layout_margins` to something smaller than `9` ?

Please let me know what you think.....

**Comment by cristinasewell** [comment]
Closing this for now after talking to @hhslepicka . 
If users don't want their PyDMEnumButton widget to expand when in a layout, they have to set a max height for it, and they should leave the margin properties of the widget to its default ones.
